icu4c/source/i18n/rbt_pars.cpp

```diff
@@ -535,15 +535,14 @@ int32_t RuleHalf::parseSection(const UnicodeString& rule, int32_t pos, int32_t l
                 // in buf, starting at offset bufSegStart.  Extract them
                 // into a string matcher, and replace them with a
                 // standin for that matcher.
-                StringMatcher* m =
-                    new StringMatcher(buf, bufSegStart, buf.length(),
-                                      segmentNumber, *parser.curData);
-                if (m == nullptr) {
+                LocalPointer<StringMatcher> m(new StringMatcher(buf, bufSegStart, buf.length(),
+                                      segmentNumber, *parser.curData), status);
+                if (m.isNull()) {
```

Comment:
- check the status that you ask the LocalPointer constructor to check and set
- more readable line breaks

```suggestion
                LocalPointer<StringMatcher> m(
                        new StringMatcher(buf, bufSegStart, buf.length(), segmentNumber, *parser.curData),
                        status);
                if (U_FAILURE(status)) {
```